ksp_plugin_adapter/time_span.cs

```diff
@@ -0,0 +1,138 @@
+    if (!Split(out int days,
+               out int hours,
+               out int minutes,
+               out double seconds)) {
+      // In case of error, saturate to the largest representable value.
+      days = int.MaxValue;
+      hours = 0;
+      minutes = 0;
+      seconds = 0;
+    }
+    var components = new List<string>();
+    if (with_leading_zeroes) {
+      components.Add(is_stock_day
+                         ? days.ToString("0000;0000")
+                         : days.ToString("000;000"));
+    } else if (days != 0) {
+      components.Add(days.ToString("0;0"));
+    }
+    if (components.Count > 0) {
+      components.Add($"{nbsp}{day_symbol}{nbsp}");
+    }
+    if (components.Count > 0 || with_leading_zeroes || hours != 0) {
+      components.Add(date_time_formatter.Day / date_time_formatter.Hour < 10
+                         ? hours.ToString("0;0")
+                         : hours.ToString("00;00"));
+      components.Add($"{nbsp}h{nbsp}");
+    }
+    if (components.Count > 0 || with_leading_zeroes || minutes != 0 ||
+        !with_seconds) {
+      components.Add(minutes.ToString("00;00"));
+      components.Add($"{nbsp}min");
+    }
+    if (with_seconds) {
+      components.Add($"{nbsp}" + seconds.ToString("00.0;00.0") + $"{nbsp}s");
+    }
+    return string.Join("", components.ToArray());
+  }
+
+  public double total_seconds => seconds_;
+
+  public static bool TryParse(string text,
+                              bool with_seconds,
+                              out PrincipiaTimeSpan time_span) {
+    time_span = new PrincipiaTimeSpan(double.NaN);
+    // Using a technology that is customarily used to parse HTML.
+    string pattern = @"^[+]?\s*(\d+)\s*" + day_symbol +
+                     @"\s*(\d+)\s*h\s*(\d+)\s*min";
+    if (with_seconds) {
+      pattern += @"\s*([0-9.,']+)\s*s$";
+    } else {
+      pattern += @"$";
+    }
+    var regex = new Regex(pattern);
+    var match = regex.Match(text);
+    if (!match.Success) {
+      return false;
+    }
+    string days = match.Groups[1].Value;
+    string hours = match.Groups[2].Value;
+    string minutes = match.Groups[3].Value;
+    string seconds = "0";
+    if (with_seconds) {
+      seconds = match.Groups[4].Value;
+    }
+    if (!int.TryParse(days, out int d) || !int.TryParse(hours, out int h) ||
+        !int.TryParse(minutes, out int min) || !double.TryParse(
+            seconds.Replace(',', '.'),
+            NumberStyles.AllowDecimalPoint | NumberStyles.AllowThousands,
+            Culture.culture.NumberFormat,
+            out double s)) {
+      return false;
+    }
+    time_span = new PrincipiaTimeSpan(d * date_time_formatter.Day +
+                                      h * date_time_formatter.Hour +
+                                      min * date_time_formatter.Minute + s);
+    return true;
+  }
+
+  public static int day_duration => date_time_formatter.Day;
+  public static string day_symbol => is_stock_day ? "d6" : "d";
```

Comment:
```C#
  public static string day_symbol =>
      hour_divides_day && !is_si_time
          ? "d" + (int)(date_time_formatter.Day / date_time_formatter.hour)
          : "d";


  private static bool hour_divides_day =>
      (int)(date_time_formatter.Day / date_time_formatter.hour) * date_time_formatter.hour ==
          date_time_formatter.Day;
  private static is_si_time =>
      date_time_formatter.Minute == 60 &&
      date_time_formatter.Hour == 60 * 60 &&
      date_time_formatter.Day == 24 * 60 * 60;
```

Reply:
Actually I made the change, and I foundthe output with JNSQ unnecessarily noisy:
```
014 d12 00 h 00 min
```
So I changed it again to only display one digit after `d`.  Thus `d3`, `d6`, `d9` but just `d` for JNSQ.